### PR TITLE
ARROW-15698: [Integration] Privatized some code in tests

### PR DESCRIPTION
**Pull request by jorgecarleitao**
* Privatized some members of the concrete Testers
* Added typing information to the `IntegrationRunner`
* Simplified some code in the `IntegrationRunner`

### CODE REVIEW

**Comment by kou** [comment]
There are too many changes to say "MINOR": https://github.com/apache/arrow/blob/master/CONTRIBUTING.md#minor-fixes

Could you open a JIRA issue?

**Comment by jorgecarleitao** [comment]
@kou , thanks. Done.

**Review by kou — APPROVED** [review]
Thanks.

+1

@kszucs @pitrou Do you want to review this before we merge this?

File: dev/archery/archery/integration/tester_cpp.py [hunk 1]
```diff
@@ -23,39 +23,46 @@
...
-    def _run(self, arrow_path=None, json_path=None, command='VALIDATE',
-             quirks=None):
-        cmd = [self.CPP_INTEGRATION_EXE, '--integration']
+    def _run(
+        self,
+        arrow_path=None,
+        json_path=None,
+        command="VALIDATE",
+        quirks=None
+    ):
```
[the remark this answers is not in the document]
Comment (jorgecarleitao) [reply dev/archery/archery/integration/tester_cpp.py:56]: quite a fight to get the linter happy.

File: dev/archery/archery/integration/tester_cpp.py [hunk 2]
```diff
@@ -23,39 +23,46 @@
...
+        cmd = [_INTEGRATION_EXE, "--integration"]
 
         if arrow_path is not None:
-            cmd.append('--arrow=' + arrow_path)
+            cmd.append("--arrow=" + arrow_path)
```
[the remark this answers is not in the document]
Comment (jorgecarleitao) [reply dev/archery/archery/integration/tester_cpp.py]: I use black on save my editor and did not even notice these changes. Reverted. Sorry for the noise.

File: dev/archery/archery/integration/runner.py [hunk 3]
```diff
@@ -18,6 +18,7 @@
 from collections import namedtuple
 from concurrent.futures import ThreadPoolExecutor
 from functools import partial
+from typing import Callable, List
```
[the remark this answers is not in the document]
Comment (jorgecarleitao) [reply dev/archery/archery/integration/runner.py]: fixed.

Without a linter it is a bit difficult to keep consistency - e.g. `archery/bot.py` has

```python
import os
import shlex
from pathlib import Path
from functools import partial
import tempfile
```

[the remark this answers is not in the document]
Comment (pitrou) [reply dev/archery/archery/integration/runner.py]: True. This is more of a best effort thing.

File: dev/archery/archery/integration/tester_cpp.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pitrou) [reply dev/archery/archery/integration/tester_cpp.py:56]: You should be able to use `archery lint --python --fix`.

**Review by pitrou — APPROVED** [review]
+1, thank you @jorgecarleitao

**Pull request closed without merging**